test/test_platforms.py

```
@@ -173,6 +174,69 @@ def test_with_build_job(self, fetch_allthethings_data):
             'platform1')
 
 
+class TestWantedBuilder(unittest.TestCase):
+
+    """Test _wanted_builder with mock data."""
+
+    @patch('mozci.platforms.fetch_allthethings_data')
+    def test_pgo(self, fetch_allthethings_data):
+        """For pgo builds it should return False as an equivalent opt build exists."""
+        fetch_allthethings_data.return_value = MOCK_ALLTHETHINGS
+        self.assertEquals(
+            _wanted_builder('Platform1 mozilla-central pgo test mochitest-1'),
+            False)
+        self.assertEquals(
+            _wanted_builder('Platform1 mozilla-aurora pgo test mochitest-1'),
+            True)
+        self.assertEquals(
+            _wanted_builder('Platform1 mozilla-inbound pgo test mochitest-1'),
+            False)
+        self.assertEquals(
+            _wanted_builder('Platform1 mozilla-beta pgo test mochitest-1'),
+            True)
+        self.assertEquals(
+            _wanted_builder('Platform1 mozilla-release pgo test mochitest-1'),
+            True)
+        self.assertEquals(
+            _wanted_builder('Platform1 mozilla-esr38 pgo test mochitest-1'),
+            True)
+        self.assertEquals(
+            _wanted_builder('Platform1 mozilla-esr45 pgo test mochitest-1'),
+            True)
+        self.assertEquals(
+            _wanted_builder('Platform1 repo pgo test mochitest-1'),
+            False)
```

Now that I think of it we need two more cases.

A new repo with only opt.
Another new repo with only pgo.

It will probably raise an exception when we try to check on a builder that does not exist on MOCK_ALLTHETHINGS.
Once _wanted_builder() calls get_buildername_metadata() with an invalid builder we should get an exception [1].
Could you please raise a MozciError here instead of None?

[1] https://github.com/mozilla/mozilla_ci_tools/blob/master/mozci/platforms.py#L478
[2] https://github.com/mozilla/mozilla_ci_tools/blob/master/mozci/platforms.py#L222